src/aznfswatchdog

```diff
@@ -138,45 +140,96 @@ reconcile_conntrack()
     done
 }
 
-vecho "Starting aznfswatchdog..."
 
-# Detect and log distro and bash version
-if [ -f /etc/centos-release ]; then
-    linux_distro=$(cat /etc/centos-release 2>&1)
-elif [ -f /etc/os-release ]; then
-    linux_distro=$(grep "^PRETTY_NAME=" /etc/os-release | awk -F= '{print $2}' | tr -d '"')
-else
-    linux_distro="Unknown"
-fi
+#
+# Delete entry from $AZ_FILES_MOUNTMAP.
+#
+ensure_az_files_mountmap_not_exist()
+{
+    (
+        flock -e 999
 
-bash_version=$(bash --version | head -n 1)
-vecho "Linux distribution: $linux_distro"
-vecho "Bash version: $bash_version"
+        #
+        # If user wants to delete the entry only if AZ_FILES_MOUNTMAP has not changed since
+        # he looked up, honour that.
+        #
+        local ifmatch="$2"
+        if [ -n "$ifmatch" ]; then
+            local mtime=$(stat -c%Y $AZ_FILES_MOUNTMAP)
+            if [ "$mtime" != "$ifmatch" ]; then
+                eecho "[$1] Refusing to remove from ${AZ_FILES_MOUNTMAP} as $mtime != $ifmatch!"
+                return 1
+            fi
+        fi
 
-# Dump NAT table once on startup in case we have reported conflicts.
-vecho "NAT table:\n$(iptables-save -t nat)"
-conntrack -L > /dev/null
+        chattr -f -i $AZ_FILES_MOUNTMAP
+        #
+        # We do this thing instead of inplace update by sed as that has a
+        # very bad side-effect of creating a new AZ_FILES_MOUNTMAP file. This breaks
+        # any locking that we dependent on the old file.
+        #
+        out=$(sed "\%^${1}$%d" $AZ_FILES_MOUNTMAP)
+        ret=$?
+        if [ $ret -eq 0 ]; then
+            #
+            # If this echo fails then AZ_FILES_MOUNTMAP could be truncated.
+            #
+            echo "$out" > $AZ_FILES_MOUNTMAP
+            ret=$?
+            out=
+            if [ $ret -ne 0 ]; then
+                eecho "*** [FATAL] AZ_FILES_MOUNTMAP may be in inconsistent state, contact Microsoft support ***"
+            fi
+        fi
```

Comment:
merge with below as `else`?

Reply:
no, i shouldn't merge with below else. I am getting 'ret' status at two places 172 and 178 line. Hence, if [ $ret -ne 0 ] can be anywhere. Also, the handling of if [ $ret -ne 0 ] is different at both the placess, so I think we shouldn't merge with below else.